src/util/time.h

```diff
@@ -24,8 +24,14 @@ void UninterruptibleSleep(const std::chrono::microseconds& n);
  * This helper is used to convert durations before passing them over an
  * interface that doesn't support std::chrono (e.g. RPC, debug log, or the GUI)
  */
-inline int64_t count_seconds(std::chrono::seconds t) { return t.count(); }
-inline int64_t count_microseconds(std::chrono::microseconds t) { return t.count(); }
+constexpr inline int64_t count_seconds(std::chrono::seconds t) { return t.count(); }
+constexpr inline int64_t count_microseconds(std::chrono::microseconds t) { return t.count(); }
+
+/**
+ * Helper to count the seconds in any std::chrono::duration type
+ */
+using SecondsDouble = std::chrono::duration<double, std::ratio<1, 1>>;
```

Comment:
nit: Would be nice to inherit the period from "upstream"


```diff
diff --git a/src/util/time.h b/src/util/time.h
index deb9ceefe2..f1b2beb0f3 100644
--- a/src/util/time.h
+++ b/src/util/time.h
@@ -30,7 +30,7 @@ constexpr inline int64_t count_microseconds(std::chrono::microseconds t) { retur
 /**
  * Helper to count the seconds in any std::chrono::duration type
  */
-using SecondsDouble = std::chrono::duration<double, std::ratio<1, 1>>;
+using SecondsDouble = std::chrono::duration<double, std::chrono::seconds::period>;
 inline double CountSeconds(SecondsDouble t) { return t.count(); }
 
 /**

Reply:
Done